src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/data/CompareUtil.java

```diff
@@ -39,9 +39,10 @@
      * Tells if the given pattern matches the string according to the implemented comparer/algorithm.
      * @param pattern the pattern to use.
      * @param str the string to match on.
+     * @param envVars the environment variables exisiting on the jenkins host.
      * @return true if the string matches the pattern.
      */
-    boolean matches(String pattern, String str);
+    boolean matches(String pattern, String str, EnvVars envVars);
```

Comment:
Darn it, I didn't see that this was an interface when I recommended the change to the signatures, but I did say deprecate the old signature and add a new one.
Since it is a public interface we can't easily tell if there are any implementations of this in other plugins and we need to try to keep binary backwards compatibility.

The simplest way I can think of at the moment would be to
 * Add back `boolean matches(String pattern, String str);` and deprecate it.
 * Add a new class: AbstractCompareUtil or something and make it an abstract class, implementing the old method signature that only calls the new method `boolean matches(String pattern, String str, EnvVars envVars);` with `null` for `envVars`.
 * Make `AntCompareUtil` et.al. extend `AbstractCompareUtil` instead of `implements CompareUtil`. They also needs to stay in this class so they keep their old names.
 * Then make all references to `CompareUtil` in this plugin reference `AbstractCompareUtil` instead.
 * Add a new file with the interface